modules/demo-magento/magento/dynamicPages/CATEGORY.vue

```diff
@@ -38,7 +37,7 @@ const [data] = await useResource(
   () => ({
     id: props.id,
     page: Number(route.query.page ?? 1),
-    pageSize: CATALOG_PAGE_SIZE * (route.query.pages || 1),
+    pageSize: route.query.pages,
```

Comment:
pageSize is size of the page (number of products per page). So passing in pages query does not make sense. Better solution would prolly be to add new argument.